sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala

```diff
@@ -1862,7 +1864,24 @@ private[lf] object SBuiltinFun {
 
       val time = getSTimestamp(args, 0)
 
-      machine.needTime(now => Control.Value(SBool(now < time)))
+      machine.needTime(now => {
+        val Time.Range(lb, ub) = machine.getTimeBoundaries
+        // We only generate a transaction trace for the last exercise command
+        val message = machine.transactionTrace(1)
+        machine.traceLog.add(message, machine.getLastLocation)(machine.loggingContext)
```

Comment:
This change adds in some debug information that should help out Daml users

Reply:
Can you given an example of where users would see these logs and how they would look like? Could they be too verbose?

Is this traceLog the same as the one used for implementing `trace`: https://docs.daml.com/daml/stdlib/Prelude.html#function-da-internal-prelude-trace-56451

That would likely be too verbose.

Reply:
Agreed, this should be a separate logger from the user-triggered trace log that logs at trace level by default. So it's disabled at debug level but you can enable it through logback